virtio-devices/src/balloon.rs

```diff
@@ -165,6 +211,55 @@ impl BalloonEpollHandler {
         Self::advise_memory_range(memory, range_base, range_len, libc::MADV_DONTNEED)
     }
 
+    fn release_memory_range_4k(
+        pbp: &mut Option<PartiallyBalloonedPage>,
+        memory: &GuestMemoryMmap,
+        pfn: u32,
+    ) -> result::Result<(), Error> {
+        let range_base = GuestAddress((pfn as u64) << VIRTIO_BALLOON_PFN_SHIFT);
+        let range_len = 1 << VIRTIO_BALLOON_PFN_SHIFT;
+
+        let page_size: u64 = get_page_size();
+        if page_size == 1 << VIRTIO_BALLOON_PFN_SHIFT {
+            return Self::release_memory_range(memory, range_base, range_len);
+        }
+
+        if pbp.is_none() {
+            *pbp = Some(PartiallyBalloonedPage::new());
+        }
+
+        let mut addr = range_base.0;
+        let mut len = range_len as i64;
+        while len > 0 {
+            if !pbp.as_ref().unwrap().pfn_match(addr) {
+                // We are trying to free memory region in a different pfn with current pbp. Flush addr
```

Comment:
Incomplete commit message? Complete with: "... Flush the bitmap and reset the base address"?

Reply:
Yeah, should be "flush pbp".